### PR TITLE
#165273440 Add timestamped comments to any article.

### PR DESCRIPTION
**Pull request by engjames**
#### What does this PR do?
- Users should be able to comment on articles

#### Description of Task to be completed
- Adds the functionality of user commenting on any article.
- Adds tests for the functionality.

#### How should this be manually tested?
- Clone the repository(https://github.com/andela/ah-frontend-invictus.git)
- Checkout to this branch (ft-comment-article-165273440)
- npm install then npm start to run the server
- Use this link to comment (http://localhost:3000/articles/comment) after successful login.

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#165273440](https://www.pivotaltracker.com/story/show/165273440)

#### Screenshots (if appropriate)
image

### CODE REVIEW

File: src/Router.js [hunk 1]
```diff
@@ -10,6 +10,8 @@ import SignupComponent from './app/containers/Signup';
 import ArticleContainer from './app/containers/ArticleContainer';
 import ProfileComponent from './app/containers/ProfileContainer';
 import editProfileComponent from './app/containers/editProfileContainer';
+import CommentContainer from './app/containers/CommentContainer';
```
Comment (njeri-ngigi) [inline src/Router.js]: commented code

File: src/app/containers/CommentContainer.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    var headers = {};
+    headers['Content-Type'] = 'application/json';
+    headers['Authorization'] = `Bearer ${localStorage.getItem('user_token')}`;
+    this.emptyBody(commentDetails);
+    postComment(commentDetails);
+  };
+
+  render () {
+    // console.log(this.state)
```
Comment (njeri-ngigi) [inline src/app/containers/CommentContainer.js]: commented code

File: src/redux/actions/CommentAction.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { toast } from 'react-toastify';
+import axios from 'axios';
+import { commentTypes } from './types';
+// import { commentUrls } from './commentUrl';
```
Comment (njeri-ngigi) [inline src/redux/actions/CommentAction.js]: commented code

File: src/redux/actions/FetchCommentAction.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import axios from 'axios';
+import { commentTypes } from './types';
+// import { commentUrls } from './commentUrl';
```
Comment (njeri-ngigi) [inline src/redux/actions/FetchCommentAction.js]: commented code

File: src/redux/actions/FetchCommentAction.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+let authToken = localStorage.getItem('user_token');
+export const fetchComments = () => dispatch => {
+  return axios.get('https://inviticus-staging.herokuapp.com/api/articles/10/comments/',
```
Comment (njeri-ngigi) [inline src/redux/actions/FetchCommentAction.js]: place this url in the urls file

File: src/app/components/Comment.jsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import '../../assets/styles/login.scss';
+import '../containers/CommentContainer';
```
Comment (njeri-ngigi) [inline src/app/components/Comment.jsx]: You've imported this but have you used it?

File: src/app/components/FetchComment.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import '../../assets/styles/login.scss';
+import '../../assets/styles/fetchcomment.scss';
+import '../containers/CommentContainer';
```
Comment (njeri-ngigi) [inline src/app/components/FetchComment.js]: Is there a reason why you've imported the whole file and not assigned it to a variable? It also seems like this isn't being used 🤷‍♀️

File: src/app/containers/CommentContainer.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+  }
+}
+
+CommentContainer.propTypes = {
+  body: PropTypes.string,
+  handleSubmit: PropTypes.func,
+  onChange: PropTypes.func,
+  CommentAction: PropTypes.func,
+  fetchComment: PropTypes.func,
+  postComment: PropTypes.func
```
Comment (njeri-ngigi) [inline src/app/containers/CommentContainer.js]: Are these props actually being used in this component?

File: src/app/components/FetchComment.js (hunk 7 above)
Comment (engjames) [reply src/app/components/FetchComment.js]: fixed it

File: src/app/components/Comment.jsx (hunk 6 above)
Comment (engjames) [reply src/app/components/Comment.jsx]: fixed this

File: src/app/containers/CommentContainer.js (hunk 8 above)
Comment (engjames) [reply src/app/containers/CommentContainer.js]: fixed it

File: src/app/containers/CommentContainer.js [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+    const { commentBody } = this.state;
+    var headers = {};
+    headers['Content-Type'] = 'application/json';
+    headers['Authorization'] = `Bearer ${localStorage.getItem('user_token')}`;
+    this.emptyBody(commentBody);
+    this.props.postComment(articleId, commentBody);
+  };
+
+  // componentWillReceiveProps (nextProps) {
+  //   if (nextProps.handleSubmit) {
```
Comment (njeri-ngigi) [inline src/app/containers/CommentContainer.js]: commented code

**Pull request merged**